assets/js/components/AdminMenuTooltip/AdminMenuTooltip.js

```diff
@@ -1,17 +1,56 @@
-import { useCallback } from '@wordpress/element';
+/**
+ * AdminMenuTooltip component.
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 
-import PropTypes from 'prop-types';
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
 
-import { useDispatch } from 'googlesitekit-data';
+/**
+ * Internal dependencies
+ */
+import { useDispatch, useSelect } from 'googlesitekit-data';
 import JoyrideTooltip from '../JoyrideTooltip';
 import { CORE_UI } from '../../googlesitekit/datastore/ui/constants';
-import { useTooltipState } from './useTooltipState';
+import { trackEvent } from '../../util';
+import useViewContext from '../../hooks/useViewContext';
 
-export function AdminMenuTooltip( { onDismiss, tooltipStateKey, ...props } ) {
+export function AdminMenuTooltip() {
+	const viewContext = useViewContext();
 	const { setValue } = useDispatch( CORE_UI );
 
-	const { rehideAdminMenu, rehideAdminSubMenu } =
-		useTooltipState( tooltipStateKey );
+	const {
+		isTooltipVisible = false,
+		rehideAdminMenu = false,
+		rehideAdminSubMenu = false,
+		onDismiss,
+		tooltipSlug,
+		...tooltipSettings
+	} = useSelect(
+		( select ) =>
+			select( CORE_UI ).getValue( 'admin-menu-tooltip' ) || {
+				isTooltipVisible: false,
+			}
+	);
```

Comment:
What's the thinking in moving all of these from props to pieces of data we store in the datastore? Are any of the props normally passed to this component things like functions or JSX that we can't serialise into JSON? In general, it's best to avoid storing that kind of stuff in the datastore in case we want to add the ability later to save things offline/serialise bits of data in the store. 🤔

Reply:
Good point, the only non serialisable item was `onDismiss` but we don't need this as we have inlined the GA events this was used for within the component. I've refactored this to use a defined set of serialisable props:

 - tooltipSlug
 - title
 - content
 - dismissLabel